include/boost/hana/hash.hpp

```diff
@@ -80,6 +80,16 @@ BOOST_HANA_NAMESPACE_BEGIN
             }
         };
 
+        template <typename T>
+        struct hash_integral_helper<T,
+            typename std::enable_if<std::is_enum<T>::value>::type
+        > {
+            template <typename X>
+            static constexpr auto apply(X const&) {
+                return hana::type_c<X>;
```

Comment:
Just taking a second look at this, is there a reason it does not normalize to `hana::type<hana::integral_constant<T, X::value>` as the others do?